### PR TITLE
Enhanced Docstrings and Examples in /ops/function.py, /ops/math.py and /ops/nn.py.

### PR DESCRIPTION
**Pull request by Faisal-Alsrheed**
**Main Changes:**

1. Improved docstrings and added examples for segment_sum, segment_max, and rsqrt functions in keras_core/ops/math.py.
2. Added docstring for network_nodes in keras_core/ops/function.py 
3. Enhanced the docstring for multi_hot function in keras_core/ops/nn.py.

My target to add very easy to understand examples.

### CODE REVIEW

**Comment by Faisal-Alsrheed** [comment]
Thank you for your patience, I'm starting to understand GitHub slowly :)

File: keras_core/ops/math.py [hunk 1]
```diff
@@ -100,11 +101,12 @@ def segment_max(data, segment_ids, num_segments=None, sorted=False):
 
     Example:
 
-    >>> data = keras_core.ops.convert_to_tensor([1, 2, 3, 4, 5, 6])
-    >>> segment_ids = keras_core.ops.convert_to_tensor([0, 1, 0, 1, 0, 1])
-    >>> segment_max(data, segment_ids)
-    array([9 12], shape=(2,), dtype=int32)
-    """
+    >>> data = keras_core.ops.convert_to_tensor([1,2,10,20,100,200])
```
Comment (fchollet) [inline keras_core/ops/math.py:104]: Likewise here?

File: keras_core/ops/math.py [hunk 2]
```diff
@@ -696,10 +698,10 @@ class Rsqrt(Operation):
 
     Example:
 
-    >>> x = keras_core.ops.convert_to_tensor([2., 3., -2.])
-    >>> rsqrt(x)
-    """
-
+   >>> data = keras_core.ops.convert_to_tensor([1.0,10.0,100.0])
```
Comment (fchollet) [inline keras_core/ops/math.py:701]: Use spaces are `,`

File: keras_core/ops/math.py [hunk 3]
```diff
@@ -696,10 +698,10 @@ class Rsqrt(Operation):
...
+   >>> keras_core.ops.rsqrt(data)
+   Array([1.        , 0.31622776, 0.1       ], dtype=float32)
```
Comment (fchollet) [inline keras_core/ops/math.py:703]: Use `array` (lowercase) and trim whitespace

File: keras_core/ops/nn.py [hunk 4]
```diff
@@ -1501,6 +1501,31 @@ def compute_output_spec(self, inputs):
     ]
 )
 def multi_hot(inputs, num_tokens, axis=-1, dtype=None):
+    """
+    Encodes integer labels as multi-hot vectors.
```
Comment (fchollet) [inline keras_core/ops/nn.py:1505]: Move this to the line above

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_core/ops/math.py [hunk 5]
```diff
@@ -48,10 +48,11 @@ def segment_sum(data, segment_ids, num_segments=None, sorted=False):
 
     Example:
 
-    >>> data = keras_core.ops.convert_to_tensor([1, 2, 3, 4, 5, 6])
-    >>> segment_ids = keras_core.ops.convert_to_tensor([0, 1, 0, 1, 0, 1])
-    >>> segment_sum(data, segment_ids)
-    array([9 12], shape=(2,), dtype=int32)
+    >>> data = keras_core.ops.convert_to_tensor([1,2,10,20,100,200])
```
[the remark this answers is not in the document]
Comment (Faisal-Alsrheed) [reply keras_core/ops/math.py:51]: The issue was that the example does not work with the JAX backend.
`ValueError: Argument 'num_segments' must be set when using the JAX backend. Received: num_segments=None
So, I have changed it to an easier-to-understand example, and now it is working with all backends.

File: keras_core/ops/math.py (hunk 1 above)
Comment (Faisal-Alsrheed) [reply keras_core/ops/math.py:104]: 1st , The issue was that the example does not work with the JAX backend.
``ValueError: Argument 'num_segments' must be set when using the JAX backend. Received: num_segments=None
`
The 2nd issue was that the wrong output ( I seems like it was a copy from segment_sum)

So, I have changed it to an easier-to-understand example, and now it is working with all backends.

File: keras_core/ops/function.py [hunk 6]
```diff
@@ -180,7 +180,7 @@ def map_graph(inputs, outputs):
 
     Returns:
         A tuple `(nodes, nodes_by_depth, operations, operations_by_depth)`.
-        - nodes: list of Node instances.
+        - network_nodes:dict mapping the unique node keys to the Node instances
```
[the remark this answers is not in the document]
Comment (Faisal-Alsrheed) [reply keras_core/ops/function.py:183]: Done

File: keras_core/ops/math.py (hunk 2 above)
Comment (Faisal-Alsrheed) [reply keras_core/ops/math.py:701]: Done

File: keras_core/ops/math.py (hunk 3 above)
Comment (Faisal-Alsrheed) [reply keras_core/ops/math.py:703]: Done

File: keras_core/ops/nn.py (hunk 4 above)
Comment (Faisal-Alsrheed) [reply keras_core/ops/nn.py:1505]: Done

**Comment by Faisal-Alsrheed** [comment]
I will be more than happy to write, enhance, and check all the docstrings. Additionally, I'll add examples that are easy to understand and compatible with all backends. Please let me know if that sounds good. @fchollet

File: keras_core/ops/math.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (fchollet) [reply keras_core/ops/math.py:51]: Sounds good -- please make sure to format the code in the `black` style, e.g. spaces after `,`

File: keras_core/ops/math.py [hunk 7]
```diff
@@ -48,10 +48,11 @@ def segment_sum(data, segment_ids, num_segments=None, sorted=False):
...
+    >>> segment_ids = keras_core.ops.convert_to_tensor([0,0,1,1,2,2])
+    >>> num_segments = 3
+    >>> keras_core.ops.segment_sum(data, segment_ids,num_segments)
+    array([  3,  30, 300], dtype=int32)
```
Comment (fchollet) [inline keras_core/ops/math.py:55]: Trim whitespace

File: keras_core/ops/math.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Faisal-Alsrheed) [reply keras_core/ops/math.py]: Great, Now i am using pip install black and this tool https://black.vercel.app/

File: keras_core/ops/math.py (hunk 7 above)
Comment (Faisal-Alsrheed) [reply keras_core/ops/math.py]: Done

**Review by Faisal-Alsrheed — COMMENTED** [review]
Done

**Review by Faisal-Alsrheed — COMMENTED** [review]
Done

**Comment by Faisal-Alsrheed** [comment]
My next step will be to check all the functions docstring and then discuss with you a standard way to write them even the order including examples and reference and so on

**Comment by fchollet** [comment]
> Great now I am using black python library and this website https://black.vercel.app/ for formatting.

Once you have installed `black` you can just run it from the terminal, e.g. `black file/path/`

File: keras_core/ops/math.py [hunk 8]
```diff
@@ -100,10 +101,11 @@ def segment_max(data, segment_ids, num_segments=None, sorted=False):
 
     Example:
 
-    >>> data = keras_core.ops.convert_to_tensor([1, 2, 3, 4, 5, 6])
-    >>> segment_ids = keras_core.ops.convert_to_tensor([0, 1, 0, 1, 0, 1])
-    >>> segment_max(data, segment_ids)
-    array([9 12], shape=(2,), dtype=int32)
+        >>> data = keras_core.ops.convert_to_tensor([1, 2, 10, 20, 100, 200])
```
Comment (fchollet) [inline keras_core/ops/math.py:104]: Unindent: the code example should have the indent as the rest of the docstring

File: keras_core/ops/math.py [hunk 9]
```diff
@@ -688,16 +690,17 @@ def stft(
 class Rsqrt(Operation):
     """Computes reciprocal of square root of x element-wise.
 
-    Args:
-        x: input tensor
+     Args:
```
Comment (fchollet) [inline keras_core/ops/math.py:693]: Same, fix indent

**Review by fchollet — COMMENTED** [review]
Thanks for the update!

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**